File: easy/3718.py
"""
3718. Smallest Missing Multiple of K

https://leetcode.com/problems/smallest-missing-multiple-of-k/

Weekly Contest 472
"""

from typing import List


class Solution:
    def missingMultiple(self, nums: List[int], k: int) -> int:
        nums_set: set[int] = set(nums)

        for fac in range(1, len(nums) + 1):
            if k * fac not in nums_set:
                return k * fac

        return k * (len(nums) + 1)

File: easy/3718.rs
// 3718. Smallest Missing Multiple of K
// https://leetcode.com/problems/smallest-missing-multiple-of-k/
// Weekly Contest 472

use std::collections::HashSet;

impl Solution {
    pub fn missing_multiple(nums: Vec<i32>, k: i32) -> i32 {
        let nums_set: HashSet<i32> = nums.clone().into_iter().collect();

        for factor in 1..=nums.len() {
            if !nums_set.contains(&(k * factor as i32)) {
                return k * factor as i32;
            }
        }

        k * (nums.len() as i32 + 1)
    }
}
